src/main/java/com/google/sps/data/Section.java

```diff
@@ -15,56 +15,112 @@
+  // Meeting times will be sorted in ascending order of time,
+  // with the first lesson being first in the list
+  private List<TimeRange> meetingTimes;
 
-    @Override
-    public boolean equals(Object other) {
-      if (other == this) {
-        return true;
-      }
+  public Section(String professor, List<TimeRange> meetingTimes) {
+    this.professor = professor;
+    this.meetingTimes = meetingTimes;
+    Collections.sort(this.meetingTimes, TimeRange.ORDER_BY_START);
 
-      if (!(other instanceof Section)) {
-        return false;
+    for (int i = 0; i < meetingTimes.size() - 1; i++) {
+      if (meetingTimes.get(i).overlaps(meetingTimes.get(i + 1))) {
+        throw new IllegalArgumentException("A section's lecture times can't overlap!");
       }
+    }
+  }
 
-      Section otherSection = (Section) other;
+  @Override
+  public boolean equals(Object other) {
+    if (other == this) {
+      return true;
+    }
 
-      return otherSection.professor.equals(this.professor)
-              && otherSection.meetingTimes.equals(this.meetingTimes);
+    if (!(other instanceof Section)) {
+      return false;
     }
 
-    @Override
-    public int hashCode() {
-        return Objects.hash(professor, meetingTimes);
+    Section otherSection = (Section) other;
+
+    return otherSection.professor.equals(this.professor) 
+      && otherSection.meetingTimes.equals(this.meetingTimes);
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(professor, meetingTimes);
+  }
+
+  public List<TimeRange> getMeetingTimes() {
+    return this.meetingTimes;
+  }
+
+  /**
+   * Function to return whether or not a section overlaps with another.
+   * 
+   * @param other the section to check for overlap with.
+   * @return If there's overlap between the current section and the other section
+   */
+  public boolean overlaps(Section other) {
+    LinkedList<TimeRange> thisTimes = new LinkedList<>(this.meetingTimes);
+    LinkedList<TimeRange> otherTimes = new LinkedList<>(other.getMeetingTimes());
+
+    TimeRange thisCurr = thisTimes.removeFirst();
+    TimeRange otherCurr = otherTimes.removeFirst();
+    TimeRange thisFirst = thisCurr;
+    TimeRange otherFirst = otherCurr;
+
+    while (!thisTimes.isEmpty() || !otherTimes.isEmpty()) {
+      if (thisCurr.overlaps(otherCurr)) {
+        return true;
+      } else if (thisCurr.end() < otherCurr.end()) {
+        if (!thisTimes.isEmpty()) {
+          thisCurr = thisTimes.removeFirst();
+        } else {
+          otherCurr = otherTimes.removeFirst();
+        }
+      } else {
+        if (!otherTimes.isEmpty()) {
+          otherCurr = otherTimes.removeFirst();
+        } else {
+          thisCurr = thisTimes.removeFirst();
+        }
+      }
     }
 
+    // Accounts for the single element list case.
+    if (thisCurr.overlaps(otherCurr)) {
+      return true;
+    }
+
+    if (thisCurr.end() > TimeRange.END_OF_WEEK && saturdayOverlapsSunday(thisCurr, otherFirst)) {
```

Comment:
These final 3 if statements could be combined & written into the final `return` statement, something like 
```
return (this.curr.overlaps(otherCurr)) 
    || (thisCurr.end > TimeRange.END_OF_WEEK && SaturdayOverlapsSunday(thisCurr, otherFirst)) 
    || (otherCurr.end > TimeRange.END_OF_WEEK && SaturdayOverlapsSunday(otherCurr, thisFirst));
```
 but I thought it would be more legible to keep it as separate if statements. Let me know if it's best practice to combine them, however!

Reply:
The more I look at this the more I'm actually preferring the consolidated version... Let me know your thoughts but I might change it to this.

Reply:
I think you could just ` return thisCurr.end() > TimeRange.END_OF_WEEK && saturdayOverlapsSunday(thisCurr, otherFirst)` since you check if its true, and return true

Reply:
I think the consolidated version is still legible, if you want to change it

Reply:
Either approach is reasonable. I think it could be made more legible by naming the chunks of your expression. Another of my comments on this PR more or less invalidates the need for this compound expression though.